accepted/00074-retracted-updates-support.md

```diff
@@ -0,0 +1,326 @@
+  package `hello-2.10-lp151.4.1.x86_64` (newer than the installed one)
+
+Operations and their outcomes:
+- listing (`zypper patches`): lists retracted patches with status `retracted`
+- searching for new versions (`zypper search -v hello`) gets us:
+  `vR | hello | package | 2.10-lp151.4.1` <--- R means retracted
+- upgrading a package (`zypper up hello`):
+  does not upgrade to a version contained in a retracted patch
+- installing specific version (`zypper in hello-2.10-lp151.4.1`):
+  **allows** installing a package contained in a retracted patch (this must be
+  taken into account, since this is the way we upgrade packages in SUMA)
+- installing retracted patch (`zypper in patch:<retracted_patch_id>`):
+  yields "patch not needed" and exit code `0`. Consequence: retracted patch
+  can't be applied to a system! Q: somebody double-check this.
+- if a system already has a patch installed, that becomes retracted, zypper has
+  no intention to fix this situation
+
+## Iteration 1 - Minimal Viable Product
+
+### Database
+The `rhnErrata` table must be updated with the `status` column of type
+`VARCHAR(32)`.
+
+### Reposync
+
+The status `retracted` is part of `update` elements in the `updatedata.xml` file
+of the repository. We need to adjust `spacewalk-repo-sync` (`errataImport.py`)
+so that it parses this attribute during repository synchronization and stores it
+in the database (`rhnErrata.status`).
+
+Q: How does SCC retract the patch? Does it modify the patch with the same id by
+   changing `status=stable` to `status=retracted` or does it somehow issue a new
+   patch?
+
+A: i think it's the former (modify), but verify that!
+   https://jira.suse.com/browse/SLE-8770
+
+Q: If the previous question==true: We need to think of consequences of modifying
+   an existing patch. Normally reposync is not designed to modify data. We need
+   to make sure reposync spots this change and maybe we also need to update some
+   caches (system available updates are cached (`rhnServerNeededCache`?)).
+
+### Java backend
+
+We need to enhance the `Errata` class with the `status` attribute, for which we
+will implement an `enum`.
+
+### Repository generating
+Enhance the `UpdateInfoWriter` class, so that it includes the `status` attribute
+in the output XML. Currently the `status` output is hardcoded to `final`.
+
+
+### UI
+
+The following pages must be updated:
+
+- Patch detail: information about the status. Given the fact we only have 2
+  states now (`retracted`/`final`), this can be a boolean information telling
+  whether the patch is retracted or not.
+  We should implement this as a checkbox, allowing the user to change
+  the attribute (although the change should be disabled for vendor patches).
+  We need to take into account the possible consequences of modifying
+  a patch (maybe regenerating some data (`rhnServerNeededCache`,
+  regenerating repodata of all involved channels)).
```

Comment:
Is this updated per our conversation on Monday?

Reply:
Yes. IIRC, there was some feedback against making this modifiable, but in the end there was no agreement not to do it. But we can always start with the simplest thing ("don't allow modification") and change it later, if there is a need for it.